{
    "file_path": "MTGO-scry-bug-test.sikuli/MTGO-scry-bug-test.py",
    "vulnerability": null,
    "source code": "\n\n\n        REGION_CONCEDE_MATCH_BUTTON.wait(\"concede_match.png\")\n        time.sleep(0.1)\n        type('\\n')\n\nclass ZeroValueDict(dict):\n    def __missing__(self, key):\n        return 0\n\ndef hash_file(file_path):\n        REGION_CONCEDE_MATCH_BUTTON.wait(\"concede_match.png\")\n        time.sleep(0.1)\n        type('\\n')\n\nclass ZeroValueDict(dict):\n    def __missing__(self, key):\n        return 0\n\ndef hash_file(file_path):\n    hasher = hashlib.md5()\n    with open(file_path, 'rb') as opened_file:\n        buf = opened_file.read()\n        hasher.update(buf)\n\n    return hasher.hexdigest()\n\ndef get_attempt_number_path(attempts):\n    return os.path.join(TEMP_PATH, 'attempt_{0}'.format(attempts))\n",
    "language": "python",
    "cwe_identifier": "CWE-328",
    "pattern_desc": "The MD5 hash function is considered insecure. Avoid using it unless explicitly needed for compatibility reasons",
    "line_number": 126,
    "line_text": "    hasher = hashlib.md5()",
    "pattern_id": "PY-R-002",
    "rule": "\\bhashlib\\.md5\\(",
    "label": 1
}